### DISCUSSION TITLE
More full-fledged alerting system

### DISCUSSION THREAD

**Discussion by dangtony98** [Infisical/infisical#5632] [Feature Request]
### Feature description
Currently, Infisical has a few alerts setup such as when a new device + IP is detected for login; however, we'd love to set up more (optional) alerts to make the platform even more helpful for users. In the future, I envision a dedicated section where organizations and members can select which types of alerts they would like to receive.

Before start implementing anything, it'd be great to crowdsource a list of ideas from y'all as to what types of alerts would be most useful. Here's a running list of possible alerts:

- [ ] Alert if an integration failed to sync.
- [ ] Alert if a new client / service-token was created to access secrets.
- [ ] Alert if a new API key was created under a user.
- [ ] Alert if the user changed/reset their password.
- [ ] Alert if a resource limit is being reached in Infisical Cloud.

### Why would it be useful?
This would be useful to inform users about various important actions happening with their account/secrets.

### Additional context
This is currently just a brainstorm; no feature development here is needed at this point.

**Comment by akhilmhdh** [comment]
## Sneak Peak

We are planning to implement a reliable queue system that will solve a lot of above issue. 😈Coming soon

**Comment by dangtony98** [comment]
Idea came up recently which is to cross-utilize events from Audit Logs V2 (coming out soon) for this alerting system. This way, users can subscribe to these events.

Every time a secret is changed, for instance, a log is created and, if an alert subscription exists, an alert will also be sent.
